x-pack/plugins/timelines/server/search_strategy/timeline/factory/events/details/query.events_details.dsl.ts

```diff
@@ -5,25 +5,43 @@
  * 2.0.
  */
 
+import { JsonObject } from '@kbn/common-utils';
 import { DocValueFields } from '../../../../../../common/search_strategy';
 
 export const buildTimelineDetailsQuery = (
   indexName: string,
   id: string,
-  docValueFields: DocValueFields[]
-) => ({
-  allowNoIndices: true,
-  index: indexName,
-  ignoreUnavailable: true,
-  body: {
-    docvalue_fields: docValueFields,
-    query: {
-      terms: {
-        _id: [id],
-      },
+  docValueFields: DocValueFields[],
+  authFilter?: JsonObject
+) => {
+  const basicFilter = {
+    terms: {
+      _id: [id],
     },
-    fields: [{ field: '*', include_unmapped: true }],
-    _source: true,
-  },
-  size: 1,
-});
+  };
+  const query =
+    authFilter != null
+      ? {
+          bool: {
+            filter: [{ ...basicFilter }, { ...authFilter }],
+          },
+        }
+      : {
+          terms: {
+            _id: [id],
+          },
+        };
+
+  return {
+    allowNoIndices: true,
+    index: indexName,
+    ignoreUnavailable: true,
+    body: {
+      docvalue_fields: docValueFields,
+      query,
+      fields: [{ field: '*', include_unmapped: true }],
+      _source: true,
+    },
+    size: 1,
+  };
+};
```

Comment:
@XavierM @angorayc I confirmed manually that we get the desired response with/without auth filter for event details, but there's work to be done in the parser to make sure it's dealing with the new alerts structure properly.